src/frontend/src/pages/FinancePage/ReimbursementRequestsSection.tsx

```diff
@@ -33,28 +38,154 @@ const ReimbursementRequestTable = ({
   const displayedReimbursementRequests =
     tabValue === 1 && allReimbursementRequests ? allReimbursementRequests : userReimbursementRequests;
 
-  const rows = displayedReimbursementRequests
-    .map(createReimbursementRequestRowData)
-    .sort((a, b) => (a.dateSubmitted > b.dateSubmitted ? -1 : 1));
+  const descendingComparator = <T,>(a: T, b: T, orderBy: keyof T) => {
+    if (b[orderBy] < a[orderBy]) {
+      return -1;
+    }
+    if (b[orderBy] > a[orderBy]) {
+      return 1;
+    }
+    return 0;
+  };
+
+  const statusDescendingComparator = (a: ReimbursementStatusType, b: ReimbursementStatusType) => {
+    const statusOrder = new Map<ReimbursementStatusType, number>([
+      [ReimbursementStatusType.PENDING_FINANCE, 1],
+      [ReimbursementStatusType.SABO_SUBMITTED, 2],
+      [ReimbursementStatusType.ADVISOR_APPROVED, 3],
+      [ReimbursementStatusType.REIMBURSED, 4],
+      [ReimbursementStatusType.DENIED, 5]
+    ]);
+
+    let bConverted = statusOrder.get(b);
+    let aConverted = statusOrder.get(a);
+
+    if (bConverted !== undefined && aConverted !== undefined) {
+      if (bConverted < aConverted) {
+        return -1;
+      }
+      if (bConverted > aConverted) {
+        return 1;
+      }
+    }
+    return 0;
+  };
+
+  const vendorDescendingComparator = (a: Vendor, b: Vendor) => {
+    if (b.name < a.name) {
+      return -1;
+    }
+    if (b.name > a.name) {
+      return 1;
+    }
+    return 0;
+  };
+
+  const submitterDescendingComparator = (a: User, b: User) => {
+    if (b.firstName < a.firstName) {
+      return -1;
+    }
+    if (b.firstName > a.firstName) {
+      return 1;
+    }
+    return 0;
+  };
+
+  const rows = displayedReimbursementRequests.map(createReimbursementRequestRowData).sort((a, b) => {
+    if (orderBy === 'vendor') {
+      return order === 'desc'
+        ? vendorDescendingComparator(a.vendor, b.vendor)
+        : -vendorDescendingComparator(a.vendor, b.vendor);
+    }
+    if (orderBy === 'status') {
+      return order === 'desc'
+        ? statusDescendingComparator(a.status, b.status)
+        : -statusDescendingComparator(a.status, b.status);
+    }
+    if (orderBy === 'submitter') {
+      return order === 'desc'
+        ? submitterDescendingComparator(a.submitter, b.submitter)
+        : -submitterDescendingComparator(a.submitter, b.submitter);
+    }
+    if (b[orderBy] === undefined) {
+      return -1;
+    }
+    return order === 'desc' ? descendingComparator(a, b, orderBy) : -descendingComparator(a, b, orderBy);
+  });
 
   const tabs = [{ label: 'My Requests', value: 0 }];
   if (canViewAllReimbursementRequests) tabs.push({ label: 'All Club Requests', value: 1 });
 
+  const handleRequestSort = (property: keyof ReimbursementRequestRow) => {
+    const isAsc = orderBy === property && order === 'asc';
+    setOrder(isAsc ? SortingOrder.descending : SortingOrder.ascending);
+    setOrderBy(property);
+  };
+
+  interface ReimbursementTableHeadCell {
```

Comment:
We wanna define all the interfaces before the main component definition like where we defined `ReimbursementRequestTableProps` interface in this file